File: src/types/order.type.ts
export type OrderType = 'VIP' | 'Normal';
export type OrderStatus = 'PENDING' | 'PROCESSING' | 'COMPLETE';

export type LogCallback = (message: string) => void;

export interface Order {
  id: number;
  type: OrderType;
  status: OrderStatus;
  createdAt: Date;
  startedAt?: Date;
  completedAt?: Date;
}

export interface OrderStats {
  totalOrders: number;
  vipOrders: number;
  normalOrders: number;
  completedOrders: number;
  pendingOrders: number;
  activeBots: number;
}

File: domain/order_type.go
package domain

type OrderType int

const (
	Normal OrderType = 0
	VIP    OrderType = 1
)

func (ot OrderType) String() string {
	switch ot {
	case Normal:
		return "Normal"
	case VIP:
		return "VIP"
	default:
		return "Unknown"
	}
}
